src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -463,6 +464,94 @@ namespace Abs {
+            }
+            return ASRUtils::EXPR(ASR::make_IntegerConstant_t(
+               al, loc, val, type));
+        } else {
+            return nullptr;
+        }
+    }
+    static inline ASR::asr_t* create_Ord(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() != 1) {
+            err("Intrinsic ord function accepts exactly 1 argument", loc);
+        }
+        ASR::ttype_t *type = ASRUtils::expr_type(args[0]);
+        if (ASRUtils::is_character(*type)) {
+            char* c = ASR::down_cast<ASR::StringConstant_t>(ASRUtils::expr_value(args[0]))->m_s;
+            if ((ASRUtils::expr_value(args[0]) != nullptr) && std::string(c).size() != 1) {
+                    err("ord() is only supported for String of length 1", args[0]->base.loc);
+                }
+        }
+        if (!ASRUtils::is_character(*type)) {
+            err("Argument of the ord function must be String of length 1",
+                args[0]->base.loc);
+        }
+        return UnaryIntrinsicFunction::create_UnaryFunction(al, loc, args, eval_Ord,
+            static_cast<int64_t>(ASRUtils::IntrinsicFunctions::Ord), 0, type);
+    }
+    static inline ASR::expr_t* instantiate_Ord(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types,
+            Vec<ASR::call_arg_t>& new_args, ASR::expr_t* compile_time_value) {
+        std::string func_name = "_lcompilers_ord_" + type_to_str_python(arg_types[0]);
+        ASR::ttype_t *return_type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc, 4, nullptr, 0));
+        if (scope->get_symbol(func_name)) {
+            ASR::symbol_t *s = scope->get_symbol(func_name);
+            ASR::Function_t *f = ASR::down_cast<ASR::Function_t>(s);
+            return ASRUtils::EXPR(ASR::make_FunctionCall_t(al, loc, s, s,
+                new_args.p, new_args.size(), expr_type(f->m_return_var),
+                compile_time_value, nullptr));
+        }
+
+        func_name = scope->get_unique_name(func_name);
+        SymbolTable *fn_symtab = al.make_new<SymbolTable>(scope);
+
+        Vec<ASR::expr_t*> args;
+        {
+            args.reserve(al, 1);
+            create_variable(arg, "x", ASR::intentType::In, ASR::abiType::Source,
+                false, fn_symtab, arg_types[0]);
+            args.push_back(al, EXPR(ASR::make_Var_t(al, loc, arg)));
+        }
+
+        create_variable(return_var, func_name, ASRUtils::intent_return_var,
+            ASR::abiType::Source, false, fn_symtab, return_type);
+
+        Vec<ASR::stmt_t*> body;
+        body.reserve(al, 1);
+        SetChar dep;
+        dep.reserve(al, 1);
+        /*
+         if (c is character) then
+            r = ord(c)
+        */
+        if (is_character(*arg_types[0])) {
+            ASR::expr_t *res;
+            res = EXPR(ASR::make_IntegerConstant_t(al, loc, int64_t(args[0]), return_type));
```

Comment:
This doesn't seem to work as expected.
I think, the compile-time value cannot be evaluated in the instantiate_ord.